spec/p2p/reactor/README.md

```diff
@@ -2,225 +2,31 @@
-
-To become a reactor, a component has first to implement the `Reactor` interface,
-then to register the implementation with the p2p layer, using the
-`Switch.AddReactor(name string, reactor Reactor)` method,
-with a global unique `name` for the reactor.
-
-The registration must happen before the node, in general, and the p2p layer,
-in particular, are started.
-In other words, there is no support for registering a reactor on a running node:
-reactors must be registered as part of the setup of a node.
-
-```abnf
-registration    = get-channels set-switch
-```
-
-The p2p layer retrieves from the reactor a list of channels the reactor is
-responsible for, using the `GetChannels()` method.
-The reactor implementation should thereafter expect the delivery of every
-message received by the p2p layer in the informed channels.
-
-The second method `SetSwitch(Switch)` concludes the handshake between the
-reactor and the p2p layer.
-The `Switch` is the main component of the p2p layer, being responsible for
-establishing connections with peers and routing messages.
-
-## Service interface
-
-A reactor must implement the [`Service`](../../../libs/service/service.go) interface,
-in particular, a startup `OnStart()` and a shutdown `OnStop()` methods:
-
-```abnf
-start           = registration on-start *peer-management on-stop
-```
-
-As part of the startup of a node, all registered reactors are started by the p2p layer.
-And when the node is shut down, all registered reactors are stopped by the p2p layer.
-Observe that the `Service` interface specification establishes that a service
-can be started and stopped only once.
-So before being started or once stopped by the p2p layer, the reactor should
-not expect any interaction.
-
-## Peer management
-
-The core of a reactor's operation is the interaction with peers or, more
-precisely, with companion reactors operating on the same channels in peers connected to the node.
-The grammar extract below represents the interaction of the reactor with a
-single peer:
+The remaining of the documentation is organized as follows:
 
-```abnf
-; Refers to a single peer, a reactor must support multiple concurrent peers
-peer-management = init-peer start-peer stop-peer
-```
+- [Reactor API](./reactor.md): documents the [`p2p.Reactor`][reactor-interface]
+  interface and specifies the behaviour of the p2p layer when interacting with
+  a reactor.
+  In other words, the interaction of the p2p layer with the protocol layer (bottom-up).
 
-The p2p layer informs all registered reactors when it establishes a connection
-with a `Peer`, using the `InitPeer(Peer)` method.
-When this method is invoked, the `Peer` has not yet been started, namely the
-routines for sending messages to and receiving messages from the peer are not running.
-This method should be used to initialize state or data related to the new
-peer, but not to interact with it.
+- [API for Reactors](./p2p-api.md): documents the interface provided by the p2p
```

Comment:
```suggestion
- [P2P API](./p2p-api.md): documents the interface provided by the p2p
```
I think P2P API better reflects the opposite direction in contrast to the "Reactor API" from above.